tests/test_representation_manager.cc

```diff
@@ -174,4 +178,94 @@ namespace rascal {
   }
 
   BOOST_AUTO_TEST_SUITE_END();
+
+  /* ---------------------------------------------------------------------- */
+
+  /* Tests specific to the spherical expansion representation
+   * TODO(max-veit) merge with the general versions above, where possible
+   */
+  BOOST_AUTO_TEST_SUITE(representation_spherical_expansion_test);
+
+  using multiple_fixtures = boost::mpl::list<
+    RepresentationFixture<StructureManagerCenters,
+                          RepresentationManagerSphericalExpansion,
+                          MultipleStructureSphericalExpansion>>;
+
+
+  BOOST_FIXTURE_TEST_CASE_TEMPLATE(
+      multiple_precompute_test, Fix, multiple_fixtures, Fix) {
+    auto& managers = Fix::managers_strict;
+    auto& representations = Fix::representations;
+    const auto& hypers = Fix::hypers;
+
+    for (auto& manager : managers) {
+      for (const auto& hyper : hypers) {
+        representations.emplace_back(manager, hyper);
+        BOOST_CHECK(representations.back().get_is_precomputed() == false);
+        representations.back().precompute();
+        BOOST_CHECK(representations.back().get_is_precomputed() == true);
+        // And now test automatic precomputation
+        representations.emplace_back(manager, hyper);
+        BOOST_CHECK(representations.back().get_is_precomputed() == false);
+        representations.back().compute();
+        BOOST_CHECK(representations.back().get_is_precomputed() == true);
+      }
+    }
+  }
+
+  //TODO(max-veit) see if this is made redundant by the general "check
+  //               representation against file" template above
+  BOOST_FIXTURE_TEST_CASE_TEMPLATE(
+      multiple_compute_test, Fix, multiple_fixtures, Fix) {
+    auto& managers = Fix::managers_strict;
+    auto& cutoffs = Fix::cutoffs;
+    auto& representations = Fix::representations;
+    const auto& filenames = Fix::filenames;
+    const auto& hypers = Fix::hypers;
+    bool verbose{true};
+
+    auto filename_it{filenames.begin()};
+    auto cutoff_it{cutoffs.begin()};
+
```

Comment:
Why not make loops within loops instead of this construction with begin/end and ++ ?

Reply:
I'm sure there was a reason originally, but maybe it can be refactored now.

Reply:
Wait, no, there really is no good way to refactor this (have a go yourself if you're not convinced)